### PR TITLE
collect and export HTTP metrics from flask

### PR DESCRIPTION
**Pull request by jharting**
https://projects.engineering.redhat.com/browse/RHCLOUD-2880

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -96,4 +97,13 @@ def set_request_id():
 
     payload_tracker.init_payload_tracker(app_config, producer=payload_tracker_producer)
 
+    # HTTP request metrics
+    if config_name != "testing":
+        PrometheusMetrics(
+            flask_app,
```
Comment (Glutexo) [inline app/__init__.py]: Maybe add `path=None` here? The _/metrics_ endpoint is already [registered](https://github.com/jharting/insights-host-inventory/blob/913ebce7125596e9b78563ae7e20a2c36c206c3c/app/__init__.py#L73), so there is no need to do it twice.

File: app/__init__.py [hunk 2]
```diff
@@ -96,4 +97,13 @@ def set_request_id():
...
+            defaults_prefix="inventory",
+            group_by="url_rule",
+            excluded_paths=["^/metrics$", "^/health$", "^/version$", "^/favicon.ico$"],
```
Comment (Glutexo) [inline app/__init__.py]: This is a regex, so the period in `^/favicon.ico$` should be escaped.
```suggestion
            excluded_paths=["^/metrics$", "^/health$", "^/version$", "^/favicon\\.ico$"],
```

File: app/__init__.py [hunk 3]
```diff
@@ -96,4 +97,13 @@ def set_request_id():
 
     payload_tracker.init_payload_tracker(app_config, producer=payload_tracker_producer)
 
+    # HTTP request metrics
+    if config_name != "testing":
```
Comment (Glutexo) [inline app/__init__.py]: These Flask metrics are disabled in testing, but the [other ones](https://github.com/jharting/insights-host-inventory/blob/913ebce7125596e9b78563ae7e20a2c36c206c3c/app/__init__.py#L73) are not. Is there a reason for that?

Comment (jharting) [reply app/__init__.py]: Yes, the metrics registry is global but the app is not. Therefore, if we register them in testing they are re-registered for every test run which causes problems

File: Pipfile [hunk 4]
```diff
@@ -17,7 +17,7 @@ flask-migrate = "~=2.5.2"
 itsdangerous = "~=1.1.0"
 request = "==2019.4.13"  # Not semver
 gunicorn = "~=19.9.0"
-connexion = {version = "~=2.3.0", extras = ["swagger-ui"]}
+connexion = {version = "~=2.3.0",extras = ["swagger-ui"]}
```
[the remark this answers is not in the document]
Comment (jharting) [reply Pipfile]: This is the formatting that pipenv prefers ¯\_(ツ)_/¯

**Pull request merged**